### PR TITLE
THRIFT-6109: Format Ruby HTTP transport endpoint labels

### PR DESCRIPTION
**Pull request by kpumuk**
Replace the malformed diagnostic label returned by `Thrift::HTTPClientTransport#to_s` with an endpoint description that includes the scheme, host, port, and path.

The label distinguishes HTTP from HTTPS while leaving URL userinfo and query parameters out of logs and exception output.

- [x] Did you create an [Apache Jira](https://issues.apache.org/jira/projects/THRIFT/issues/) ticket? [THRIFT-6109](https://issues.apache.org/jira/browse/THRIFT-6109)
- [x] If a ticket exists: Does your pull request title follow the pattern "THRIFT-NNNN: describe my issue"?
- [x] Did you squash your changes to a single commit?  (not required, but preferred)
- [x] Did you do your best to avoid breaking changes?  If one was needed, did you label the Jira ticket with "Breaking-Change"?
- [ ] If your change does not involve any code, include `[skip ci]` anywhere in the commit message to free up build resources.

### CODE REVIEW

File: lib/rb/lib/thrift/transport/http_client_transport.rb [hunk 1]
```diff
@@ -62,7 +62,7 @@ def flush
     end
 
     def to_s
-      "@{self.url}"
+      "#{@url.scheme}(#{@url.host}:#{@url.port}#{@url.path})"
     end
```
[the remark this answers is not in the document]
Comment (kpumuk) [reply lib/rb/lib/thrift/transport/http_client_transport.rb:68]: Good catch

File: lib/rb/lib/thrift/transport/http_client_transport.rb [hunk 2]
```diff
@@ -62,7 +62,9 @@ def flush
...
 
     def to_s
-      "@{self.url}"
+      path = @url.path.to_s
+      path = '/' if path.empty?
+      "#{@url.scheme}(#{@url.host}:#{@url.port}#{path})"
```
[the remark this answers is not in the document]
Comment (kpumuk) [reply lib/rb/lib/thrift/transport/http_client_transport.rb:67]: OMG, how do you come up with this edge cases, and why didn't you give them all at once to me? 😂

[the remark this answers is not in the document]
Comment (kpumuk) [reply lib/rb/lib/thrift/transport/http_client_transport.rb:67]: Added a test, we actually format the host correctly for ipv6

**Comment by Jens-G** [comment]
### Code review

No issues found. Checked for bugs, AGENTS.md compliance, and security implications.

🤖 Generated with [Claude Code](https://claude.ai/code)
